qiskit/transpiler/passes/scheduling/alap.py

```diff
@@ -84,9 +84,16 @@ def pad_with_delays(qubits: List[int], until, unit) -> None:
             # validate node.op.duration
             if node.op.duration is None:
                 indices = [bit_indices[qarg] for qarg in node.qargs]
-                raise TranspilerError(
-                    f"Duration of {node.op.name} on qubits {indices} is not found."
-                )
+                if node.op.name in dag.calibrations:
+                    if dag.calibrations[node.op.name] is not None:
+                        for key in dag.calibrations[node.op.name]:
+                            if tuple(indices) in key:
```

Comment:
how does this differ from `has_calibration_for`? If you're already doing the work, why not move this code there?

Reply:
Found the problem was in fact that the gate parameters need to be cast to `float` when the calibration is added. After fixing that, `.has_calibration_for()` runs just fine!